### PR TITLE
Phase2-hgx220 Convert DDHGCalEEAlgo to DD4Hep format

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Convert DDHGCalEEAlgo from DDD to DD4Hep scenario (Raman Shegal)

#### PR validation:

Tested with scripts in Geometry/HGCalCommonData/test/python

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+                                        << " of dimensions " << rinB << ", " << routF << ", " << hthick
+                                        << ", 0.0, 360.0 and position " << glog.name() << " number " << copy << ":"
+                                        << layerCenter_[copy - 1];
+#endif
+          PositionSensitive(ctxt, e, glog, rinB, routF, zz, layerSense_[ly], layerCenter_[copy - 1]);  //, cpv);
+        }
+
+        dd4hep::Position r1(0, 0, zz);
+        dd4hep::Rotation3D rot;
+        mother_.placeVolume(glog, copy, dd4hep::Transform3D(rot, r1));
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc]: According to #27997, it looks like the `Rotation3D` is not needed. Have you seen that PR?

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+#ifdef EDM_ML_DEBUG
+            if (iu > iumAll)
+              iumAll = iu;
+            if (iv > ivmAll)
+              ivmAll = iv;
+            ++nin;
+#endif
+
+            dd4hep::Position tran(xpos, ypos, 0.0);
+            dd4hep::Rotation3D rotation;
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc]: Unneeded `Rotation3D`?

Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc]: Taken care of that

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc]: Yes - now I removed this

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**